docs/app-dev/indexing-transactions.md

```diff
@@ -4,9 +4,14 @@ order: 6
 
 # Indexing Transactions
 
-Tendermint allows you to index transactions and later query or subscribe to their results.
-
-Events can be used to index transactions and blocks according to what happened
+Tendermint allows you to index transactions and blocks and later query or
+subscribe to their results. Both transactions and blocks are indexed by their
+respective events. However, transactions are also indexed by a primary key which
+includes the transaction hash and maps to and stores the corresponding `TxResult`.
+Blocks are indexed by a primary key which includes the block height and maps to
+and stores the block height, i.e. the block itself is never stored.
```

Comment:
I'd replace i.e. with a colon at the end. 

also I'm not clear what "transactions and blocks are indexed by their respective events," actually means.

Reply:
why `;` over `,`? Just curious.

> transactions and blocks are indexed by their respective events

transactions are indexed by `TxResult.Events` and blocks are indexed by `{Begin|End}BlockResponse.Events`

Reply:
just personal preference/style. 

I think more explicit is usually good.